tools/driver/driver.cpp

```diff
@@ -72,9 +75,10 @@ static bool shouldRunAsSubcommand(StringRef ExecName,
                                   SmallVectorImpl<const char *> &Args) {
   assert(Args.size() > 0);
 
-  // If we are not run as 'swift', don't do anything special. This doesn't work
-  // with symlinks with alternate names, but we can't detect 'swift' vs 'swiftc'
-  // if we try and resolve using the actual executable path.
+  // We only recognize subcommands in the form of 'swift <name-of-subcommand>'.
+  // If we are being run as an executable named something other than 'swift',
+  // we do not run a subcommand. This means we do not recognize symlinks with
+  // alternate names, for example 'swiftc build', as subcommands.
```

Comment:
That's not what the original text was getting at. We *do* want to distinguish "swift" vs. "swiftc", but ideally we *wouldn't* distinguish "swift" vs. "swift-4.1" (or similar).